packages/amplify_core/lib/src/types/analytics/hub/analytics_hub_event.dart

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@template amplify_common.hub.analytics_hub_event_type}
+/// Hub Event types for the Analytics category.
+/// {@endtemplate}
+enum AnalyticsHubEventType {
+  /// {@macro amplify_common.hub.analytics_hub_event_pinpoint_endpoint_id_set}
+  pinpointEndpointIdSet('PINPOINT_ENDPOINT_ID_CONFIGURED');
+
+  /// The event name.
+  final String eventName;
+
+  /// {@macro amplify_common.hub.analytics_hub_event_type}
+  const AnalyticsHubEventType(this.eventName);
+}
+
+/// The base class for hub events of the Analytics category.
+///
+/// One of [AnalyticsHubEvent.values].
+class AnalyticsHubEvent extends HubEvent<AnalyticsHubEventPayload>
+    with AWSEquatable<AnalyticsHubEvent>, AWSDebuggable {
+  AnalyticsHubEvent._(
+    this.type, {
+    AnalyticsHubEventPayload? payload,
+  }) : super(type.eventName, payload: payload);
+
+  /// {@macro amplify_common.hub.analytics_hub_event_type}
+  final AnalyticsHubEventType type;
+
+  /// Emitted when pinpoint id set on app start
+  AnalyticsHubEvent.pinpointEndpointIdSet(AnalyticsHubEventPayload payload)
+      : this._(AnalyticsHubEventType.pinpointEndpointIdSet, payload: payload);
+
+  @override
+  List<Object?> get props => [type, payload];
+
+  @override
+  String get runtimeTypeName => 'AnalyticsHubEvent';
+}
+
+class AnalyticsHubEventPayload {
+  const AnalyticsHubEventPayload({required this.pinpointEndpointId});
+
+  final String pinpointEndpointId;
```

Comment:
Hub events should not mention AWS services

```suggestion
  final String endpointId;
```